core/conversion/var/Var.cpp

```diff
@@ -94,10 +94,18 @@ nvinfer1::ITensor* Var::ITensorOrFreeze(ConversionCtx* ctx) {
       "Requested either IValue containing a Tensor, or ITensor, however Var type is " << type_name());
 
   nvinfer1::ITensor* out;
-
+  auto weights = converters::Weights();
   if (isIValue()) {
-    auto weights = converters::Weights(ctx, ptr_.ivalue->toTensor());
-
+    auto tensor = ptr_.ivalue->toTensor();
+    if (tensor.scalar_type() == at::kLong && ctx->settings.truncate_long_and_double) {
+      weights = converters::Weights(ctx, tensor.toType(at::kInt));
+      LOG_WARNING("Truncate kLong to kInt for IValue");
```

Comment:
Warning: Truncating weight (constant in the graph) from kLong to kInt to indicate that only constants are affected.